packages/blocks/src/store/process-block-type.js

```diff
@@ -75,24 +75,62 @@ function stabilizeSupports( rawSupports ) {
 	// custom block plugins that rely on immutable supports are not affected.
 	// See: https://github.com/WordPress/gutenberg/pull/66849#issuecomment-2463614281
 	const newSupports = {};
-	for ( const [ key, value ] of Object.entries( rawSupports ) ) {
-		if (
-			key === 'typography' &&
-			typeof value === 'object' &&
-			value !== null
-		) {
-			newSupports.typography = Object.fromEntries(
-				Object.entries( value ).map(
-					( [ typographyKey, typographyValue ] ) => [
-						TYPOGRAPHY_SUPPORTS_EXPERIMENTAL_TO_STABLE[
-							typographyKey
-						] || typographyKey,
-						typographyValue,
-					]
-				)
+
+	for ( const [ support, config ] of Object.entries( rawSupports ) ) {
+		// Add the current support's config as is if it does not need stabilization.
+		if ( ! EXPERIMENTAL_TO_STABLE_KEYS[ support ] ) {
+			newSupports[ support ] = config;
+			continue;
+		}
+
+		// Stabilize the support's key if needed e.g. __experimentalBorder => border.
+		if ( typeof EXPERIMENTAL_TO_STABLE_KEYS[ support ] === 'string' ) {
+			const stabilizedKey = EXPERIMENTAL_TO_STABLE_KEYS[ support ];
+
+			// If there's no stabilized key present, just use the config as is.
+			if ( ! Object.hasOwn( rawSupports, stabilizedKey ) ) {
+				newSupports[ stabilizedKey ] = config;
+				continue;
+			}
+
+			// Determine the insertion order for both key.
```

Comment:
Tiny nit: I like how the JS and PHP implementations match in terms of checking the order in which keys were added. The PHP implementation has slightly clearer comments. Would it be worth adding those two to the JS implementation? E.g.

```
// Determine the order of keys, so the last defined can be preferred.
```

And

```
// Update support config, prefer the last defined value.
```

My main thinking here is that it mightn't be clear to our future selves why we've gone with ensuring the most recently updated key wins out, so the extra comment might help with clarity. Not a big issue, though!

Reply:
Good idea, these bits were iterated on a little out of sync with each other, so they were due to be brought in line.

Reply:
Done: https://github.com/WordPress/gutenberg/pull/66918/commits/611de98df6a3345f6bb2b6810cde873b92d56b74